### PR TITLE
Delete users and orgs

### PR DESCRIPTION
**Pull request by docmerlin**
_Briefly describe your proposed changes:_
_What was the problem?_
 Tasks for entire users and orgs were not deletable in the tasks backend
_What was the solution?_
Make a method for deleting users and one for deleting orgs in the backend.Store interface

### CODE REVIEW

File: task/backend/inmem_store.go [hunk 1]
```diff
@@ -236,3 +236,54 @@ func (s *inmem) FinishRun(ctx context.Context, taskID, runID platform.ID) error
...
+func getOrg(st StoreTask) platform.ID {
+	return st.Org
+}
+
+// DeleteOrg syncronously deletes an org and all their tasks from a bolt store.
+func (s *inmem) DeleteOrg(ctx context.Context, id platform.ID) error {
+	return s.delete(ctx, id, getOrg)
+}
+
+// DeleteUser syncronously deletes a user and all their tasks from a bolt store.
```
Comment (mark-rushakoff) [inline task/backend/inmem_store.go]: Should be "from an in-mem store." And synchronously has an `h` in it.

File: task/backend/store.go [hunk 2]
```diff
@@ -38,6 +45,12 @@ type Store interface {
 	// FinishRun removes runID from the list of running tasks and if its `now` is later then last completed update it.
 	FinishRun(ctx context.Context, taskID, runID platform.ID) error
 
+	// DeleteOrg if the org does not exist, nil is returned
```
Comment (mark-rushakoff) [inline task/backend/store.go]: Godoc should be complete sentences, with a period. https://blog.golang.org/godoc-documenting-go-code

Please fix this sentence and add punctuation to the other godoc introduced in this file.

File: task/backend/storetest/storetest.go [hunk 3]
```diff
@@ -469,3 +474,86 @@ from(db:"test") |> range(start:-1h)`
...
+
+func testStoreDeleteUser(t *testing.T, create CreateStoreFunc, destroy DestroyStoreFunc) {
+	s := create(t)
+	defer destroy(t, s)
+	ids := createABunchOFTasks(t, s,
+		func(u, _ uint64) bool {
+			if u == 0 {
+				return true
+			}
+			return false
```
Comment (mark-rushakoff) [inline task/backend/storetest/storetest.go]: This can be simplified to `return u == 0`.

File: task/backend/storetest/storetest.go [hunk 4]
```diff
@@ -469,3 +474,86 @@ from(db:"test") |> range(start:-1h)`
...
+
+func testStoreDeleteOrg(t *testing.T, create CreateStoreFunc, destroy DestroyStoreFunc) {
+	s := create(t)
+	defer destroy(t, s)
+	ids := createABunchOFTasks(t, s,
+		func(_, o uint64) bool {
+			if o == 0 {
+				return true
+			}
+			return false
```
Comment (mark-rushakoff) [inline task/backend/storetest/storetest.go]: This can be simplified to `return o == 0`.

File: task/backend/storetest/storetest.go [hunk 5]
```diff
@@ -469,3 +474,86 @@ from(db:"test") |> range(start:-1h)`
...
+		},
+	)
+	org := make(platform.ID, 8)
+	err := s.DeleteOrg(context.TODO(), org)
+	if err != nil {
+		t.Fatal(err)
+	}
+	for i := range ids {
+		task, err := s.FindTaskByID(context.TODO(), ids[i])
```
Comment (mark-rushakoff) [inline task/backend/storetest/storetest.go]: The godoc for `context` says to use context.Background for tests: https://golang.org/pkg/context/#Background

File: task/backend/bolt/bolt.go [hunk 6]
```diff
@@ -499,3 +499,119 @@ func padID(id platform.ID) platform.ID {
...
+		ub := b.Bucket(usersPath).Bucket(userID)
+		if ub == nil {
+			return backend.ErrUserNotFound
+		}
+		c := ub.Cursor()
+		i := 0
+		for k, _ := c.First(); k != nil; k, _ = c.Next() {
+			i++
+			// check for cancelation every 128 tasks deleted
+			if i&(256-1) == 0 {
```
Comment (mark-rushakoff) [inline task/backend/bolt/bolt.go]: The comment says 128, but the value differs. IMO `0xff` is slightly clearer than `256-1` for a bitwise operation, but I'm fine with it as-is.

File: task/backend/storetest/storetest.go (hunk 3 above)
Comment (docmerlin) [reply task/backend/storetest/storetest.go]: doh!

File: task/backend/storetest/storetest.go (hunk 4 above)
Comment (docmerlin) [reply task/backend/storetest/storetest.go]: Again, doh

**Pull request merged**